### PR TITLE
Make the AD tests work again

### PR DESCRIPTION
**Pull request by vsedmik**
Proposed changes:
- moved the setup to the fixture completely
- added `--membership-software=samba` in enroll_ad fixture
- added changes in sssd.conf and auth_gssapi.conf per other discussion with DEVs

### CODE REVIEW

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ldap_authentication.py::test_single_sign_on_ldap_ad_server

File: pytest_fixtures/component/satellite_auth.py [hunk 1]
```diff
@@ -451,3 +452,32 @@ def enroll_ad_and_configure_external_auth(request, ad_data, target_sat):
...
+    target_sat.execute('satellite-installer --foreman-ipa-authentication=true', timeout='20m')
+
+    # add foreman ad_gp_map_service (BZ#2117523)
+    line_number = int(
+        target_sat.execute(
+            "awk -v search='domain/' '$0~search{print NR; exit}' /etc/sssd/sssd.conf"
+        ).stdout
+    )
+    target_sat.execute(
+        f'sed -i "{line_number + 1}i ad_gpo_map_service = +foreman" /etc/sssd/sssd.conf'
```
Comment (mshriver) [inline pytest_fixtures/component/satellite_auth.py:466]: Consider for sanity's sake logging the file contents after modification, that way if/when there's a test failure you can see this update was successful.

File: tests/foreman/destructive/test_ldap_authentication.py [hunk 2]
```diff
@@ -248,37 +248,19 @@ def test_single_sign_on_ldap_ad_server(
...
-        result = target_sat.execute(curl_command)
-        assert 'redirected' in result
-        assert f'{target_sat.url}/hosts' in result
-    finally:
-        # resetting the settings to default for external auth
-        target_sat.execute('satellite-installer --foreman-ipa-authentication=false', timeout=800000)
-        target_sat.execute('satellite-maintain service restart', timeout=300000)
+    # create the kerberos ticket for authentication
+    target_sat.execute(f'echo {settings.ldap.password} | kinit {settings.ldap.username}')
+    if is_open('BZ:1941997'):
```
Comment (mshriver) [inline tests/foreman/destructive/test_ldap_authentication.py:257]: Does this test end up providing coverage for the BZ? add to `:bz:` token if it does.

Comment (vsedmik) [reply tests/foreman/destructive/test_ldap_authentication.py:257]: Yes it covers the BZ too. Good point, I'll add it. :+1:

**Comment by vsedmik** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_ldap_authentication.py::test_single_sign_on_ldap_ad_server

**Review by omkarkhatavkar — APPROVED** [review]

**Review by synkd — APPROVED** [review]

**Review by latran — APPROVED** [review]

**Review by ColeHiggins2 — APPROVED** [review]

File: pytest_fixtures/component/satellite_auth.py [hunk 3]
```diff
@@ -451,3 +452,32 @@ def enroll_ad_and_configure_external_auth(request, ad_data, target_sat):
     target_sat.execute(f'echo {settings.ldap.password} | {token_command}')
     target_sat.execute('chown root.apache /etc/httpd/conf/http.keytab')
     target_sat.execute('chmod 640 /etc/httpd/conf/http.keytab')
+
+    # enable the foreman-ipa-authentication feature
+    target_sat.execute('satellite-installer --foreman-ipa-authentication=true', timeout='20m')
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/satellite_auth.py:457]: Would encourage to use InstallerCommand class here and verify if its enabled 
```suggestion
    result = target_sat.install(
        InstallerCommand('foreman-ipa-authentication true')
    )
    assert result.status == 0
```

**Review by Gauravtalreja1 — COMMENTED** [review]
Ack, with one suggestion

File: pytest_fixtures/component/satellite_auth.py (hunk 3 above)
Comment (vsedmik) [reply pytest_fixtures/component/satellite_auth.py]: Updated.

**Review by Gauravtalreja1 — APPROVED** [review]

**Pull request merged**